tcms/testcases/views.py

```diff
@@ -603,120 +604,62 @@ def printable(request,  # pylint: disable=missing-permission-required
     return render(request, template_name, context_data)
 
 
-def update_testcase(request, test_case, tc_form):
-    """Updating information of specific TestCase
-
-    This is called by views.edit internally. Don't call this directly.
-
-    Arguments:
-    - test_case: instance of a TestCase being updated
-    - tc_form: instance of django.forms.Form, holding validated data.
-    """
-
-    # TODO: this entire function doesn't seem very useful
-    # part if it was logging the changes but now this is
-    # done by simple_history. Should we remove it ???
-    # Modify the contents
-    fields = ['summary',
-              'case_status',
-              'category',
-              'priority',
-              'notes',
-              'text',
-              'is_automated',
-              'script',
-              'arguments',
-              'extra_link',
-              'requirement']
-
-    for field in fields:
-        if getattr(test_case, field) != tc_form.cleaned_data[field]:
-            setattr(test_case, field, tc_form.cleaned_data[field])
-    try:
-        if test_case.default_tester != tc_form.cleaned_data['default_tester']:
-            test_case.default_tester = tc_form.cleaned_data['default_tester']
-    except ObjectDoesNotExist:
-        pass
-
-    test_case.save()
-
-
-@permission_required('testcases.change_testcase')
-def edit(request, case_id):
-    """Edit case detail"""
-    try:
-        test_case = TestCase.objects.select_related().get(case_id=case_id)
-    except ObjectDoesNotExist:
-        raise Http404
+@method_decorator(permission_required('testcases.change_testcase'), name='dispatch')
+class EditTestCaseView(UpdateView):
+    model = TestCase
+    template_name = 'testcases/mutable.html'
+    form_class = NewCaseForm
 
-    test_plan = plan_from_request_or_none(request)
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
 
-    if request.method == "POST":
-        form = NewCaseForm(request.POST)
-        if request.POST.get('product'):
-            form.populate(product_id=request.POST['product'])
-        elif test_plan:
-            form.populate(product_id=test_plan.product_id)
+        if 'notify_form' not in kwargs:
```

Comment:
The key 'notify_form' will be included, if form_invalid method is called. That way the validation errors will be shown to the user. Although, I think that the usage of two forms for that kind of view is not good idea, please advice.